cpp/nvtx3.hpp

```diff
@@ -502,22 +548,39 @@
  *
  */
 
-/**
- * @brief Enables the use of constexpr when support for C++14 relaxed constexpr
+/* Temporary helper #defines, removed with #undef at end of header */
+
+/* Within this header, nvtx3::NVTX3_VERSION_NAMESPACE resolves to nvtx3::vX,
+ * where "X" is the major version number. */
+#define NVTX3_CONCAT(A, B) A##B
+#define NVTX3_NAMESPACE_FOR(VERSION) NVTX3_CONCAT(v, VERSION)
+#define NVTX3_VERSION_NAMESPACE NVTX3_NAMESPACE_FOR(NVTX3_CPP_VERSION_MAJOR)
+
+/* Enables the use of constexpr when support for C++14 relaxed constexpr
  * is present.
  *
  * Initializing a legacy-C (i.e., no constructor) union member requires
  * initializing in the constructor body. Non-empty constexpr constructors
  * require C++14 relaxed constexpr.
- *
  */
 #if __cpp_constexpr >= 201304L
 #define NVTX3_RELAXED_CONSTEXPR constexpr
 #else
 #define NVTX3_RELAXED_CONSTEXPR
 #endif
 
+/* Implementation sections, enclosed in guard macros for each minor version */
+
+#ifndef NVTX3_CPP_DEFINITIONS_V1_0
+#define NVTX3_CPP_DEFINITIONS_V1_0
+
 namespace nvtx3 {
+
+#if defined(NVTX3_INLINE_THIS_VERSION)
+inline
+#endif
+namespace NVTX3_VERSION_NAMESPACE {
```

Comment:
Assuming changes from above.
```suggestion
namespace NVTX3_NAMESPACE {
```

Reply:
See comments in above blurb -- resolving this one